src/mlpack/tests/main_tests/hmm_loglik_test.cpp

```diff
@@ -0,0 +1,165 @@
+/**
+ * @file hmm_loglik_test.cpp
+ * @author Daivik Nema
+ *
+ * Test mlpackMain() of hmm_loglik_main.cpp
+ */
+#include <string>
+
+#define BINDING_TYPE BINDING_TYPE_TEST
+static const std::string testName = "HMMLoglik";
+
+#include <mlpack/core.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include "test_helper.hpp"
+#include <mlpack/methods/hmm/hmm_model.hpp>
+#include <mlpack/methods/hmm/hmm.hpp>
+#include <mlpack/methods/hmm/hmm_loglik_main.cpp>
+
+#include <boost/test/unit_test.hpp>
+#include "../test_tools.hpp"
+
+using namespace mlpack;
+
+struct HMMLoglikTestFixture
+{
+ public:
+  HMMLoglikTestFixture()
+  {
+    // Cache in the options for this program.
+    CLI::RestoreSettings(testName);
+  }
+
+  ~HMMLoglikTestFixture()
+  {
+    // Clear the settings.
+    bindings::tests::CleanMemory();
+    CLI::ClearSettings();
+  }
+};
+
+BOOST_FIXTURE_TEST_SUITE(HMMLoglikMainTest, HMMLoglikTestFixture);
+
+struct Init
+{
+  template<typename HMMType>
+  static void Apply(HMMType& hmm, vector<mat>* trainSeq)
+  {
+    const size_t states = 2;
+
+    // Create the initialized-to-zero model.
+    Create(hmm, *trainSeq, states);
+
+    // Initializing the emission distribution depends on the distribution.
+    // Therefore we have to use the helper functions.
+    RandomInitialize(hmm.Emission());
+  }
```

Comment:
Since these appear in multiple files, I think it might make sense to declare a file `hmm_test_utils.hpp` in this test directory and move these structures there.